src/libraries/System.Security.Cryptography.Encoding/src/System/Security/Cryptography/PemEncoding.cs

```diff
@@ -0,0 +1,427 @@
+        /// data location of the encoded data.
+        /// </returns>
+        public static PemFields Find(ReadOnlySpan<char> pemData)
+        {
+            if (!TryFind(pemData, out PemFields fields))
+            {
+                throw new ArgumentException(SR.Argument_PemEncoding_NoPemFound, nameof(pemData));
+            }
+            return fields;
+        }
+
+        /// <summary>
+        /// Finds the first PEM-encoded data.
+        /// </summary>
+        /// <param name="pemData">
+        /// A span containing the PEM encoded data.
+        /// </param>
+        /// <param name="fields">
+        /// When this method returns <c>true</c>, the found <see cref="PemFields"/> structure
+        /// that contains the location, label, and data location of the encoded data.
+        /// </param>
+        /// <returns>
+        /// <c>true</c> if PEM encoded data was found; otherwise <c>false</c>.
+        /// </returns>
+        public static bool TryFind(ReadOnlySpan<char> pemData, out PemFields fields)
+        {
+            // Check for the minimum possible encoded length of a PEM structure
+            // and exit early if there is no way the input could contain a well-formed
+            // PEM.
+            if (pemData.Length < Preeb.Length + Ending.Length * 2 + Posteb.Length)
+            {
+                fields = default;
+                return false;
+            }
+
+            int preebLinePosition = 0;
+            while (TryReadNextLine(pemData, ref preebLinePosition, out Range lineRange))
+            {
+                ReadOnlySpan<char> line = pemData[lineRange];
+                int preebIndex = line.IndexOf(Preeb);
+
+                if (preebIndex == -1 ||
+                    (preebIndex > 0 && !line[..preebIndex].IsWhiteSpace())) // can only be preceeded by whitespace
+                {
+                    continue;
+                }
+
+                int preebEndingIndex = line[(preebIndex + Preeb.Length)..].IndexOf(Ending);
+
+                if (preebEndingIndex == -1)
+                {
+                    continue;
+                }
+
+                (int preebOffset, _) = lineRange.GetOffsetAndLength(pemData.Length);
+                int preebStartIndex = preebOffset + preebIndex;
+                int startLabelIndex = preebStartIndex + Preeb.Length;
+                int endLabelIndex = startLabelIndex + preebEndingIndex;
+                Range labelRange = startLabelIndex..endLabelIndex;
+                ReadOnlySpan<char> label = pemData[labelRange];
+
+                if (!IsValidLabel(label))
+                {
+                    continue;
+                }
+
+                ReadOnlySpan<char> posteb = string.Concat(Posteb, label, Ending);
+                Range postebLineRange = lineRange;
+                int postebLinePosition = preebLinePosition;
+
+                // in lax decoding a posteb may appear on the same line as the preeb.
```

Comment:
This wasn't actually immediately clear to me from the RFC, if we even want to support single-line PEMs.

If not, this code can get a little simpler but I suspect someone will wish it "just worked".

Reply:
> Presumably, removing all newline processing (and ReadLine) and just using IndexOf to find preeb/posteb would actually make things simpler?

It still behaved as lax, but agree that it isn't necessary to handle line-by-line. I will address. It seems I may have misunderstood the whitespace requirement around the preeb / posteb for lax.